Classes&Inheritance.js

```diff
@@ -0,0 +1,130 @@
+"use strict";
+
+//Parent class
+function Builder(startValue) {
+    this.value = startValue;
+}
+
+//Common methods for str and int
+Builder.prototype.get = function () {
+    return this.value;
+}
+Builder.prototype.plus = function () { }
+Builder.prototype.minus = function () { }
+Builder.prototype.multiply = function () { }
+Builder.prototype.divide = function () { }
+
+//Unique methods for string
+Builder.prototype.remove = function (letter) {
+    var result = "";
+    for (var i = 0; i < this.value.length; i++) {
+        if (this.value[i] !== letter) {
+            result += this.value.substring(i, i + 1);
+        }
+    }
+    this.value = result;
+    return this;
+}
+
+Builder.prototype.sub = function (start, length) {
+    this.value = this.value.substr(start, length);
+    return this;
+}
+
+//Unique methods for int
+
+Builder.prototype.mod = function (number) {
+    this.value %= number;
+    return this;
+}
+
+// Child Class - StrBuilder in ES5
+function StringBuilder(startValue) {
+    Builder.call(this, startValue || "");
+}
+
+StringBuilder.prototype = Object.create(Builder.prototype);
+StringBuilder.prototype.constructor = StringBuilder;
+
+StringBuilder.prototype.plus = function (arg) {
+    var strings = [arg];
+    for (var i = 0; i < strings.length; i++) {
+        this.value += strings[i];
+    }
+    return this;
+}
```

Comment:
```suggestion
StringBuilder.prototype.plus = function () {
    for (var i = 0; i < arguments.length; i++) {
        this.value += arguments[i];
    }
    return this;
}
```
Need to clarify this method. I think, there is always strings.length = 1. You can use arguments here.

Reply:
Done.